core/src/main/java/brooklyn/entity/rebind/PeriodicDeltaChangeListener.java

```diff
@@ -383,7 +385,7 @@ protected void persistNowInternal(boolean alreadyHasMutex) {
                 deltaCollector = new DeltaCollector();
             }
             
-            if (LOG.isDebugEnabled()) LOG.debug("Checkpointing delta of memento: "
+            if (shouldLogCheckpoint() && LOG.isDebugEnabled()) LOG.debug("Checkpointing delta of memento: "
```

Comment:
Swap the order of the if conditions for cheaper short-circuiting when `isDebugEnabled` is false.